test/mitmproxy/addons/test_script.py

```diff
@@ -84,6 +85,25 @@ def test_reload(self, tmpdir):
                 time.sleep(0.1)
             tctx.master.has_log("Loading")
 
+    def test_deletion(self):
+        with open(tutils.test_data.path("mitmproxy/data/addonscripts/") + "/dummy.py", 'w') as f:
+            f.write("\n")
+        with taddons.context() as tctx:
+            sc = script.Script(
+                tutils.test_data.path("mitmproxy/data/addonscripts/dummy.py")
+            )
+            tctx.master.addons.add(sc)
+            tctx.configure(sc)
+
+            sl = script.ScriptLoader()
```

Comment:
`ScriptLoader` addon is the parent addon which manages the `Script` addons. 
Once the script file is deleted the script addon should be removed from both `master.addons.lookup` and `ScriptLoader.addons` 
I used `ScriptLoader` so as to assert that the script is removed from it's addons as well